### DISCUSSION TITLE
Passing arguments to Alpine.data()

### DISCUSSION THREAD

**Discussion by jrmajor** [alpinejs/alpine#1500] [1. Help] [answered]
With global function data providers I used to pass encoded data from backend like this:

```blade
<div x-data="dataProvider(@json($data))"></div>
```

I can't figure out how to do it with providers registered with `Alpine.data()`. Is it possible to pass arguments to them?

**Comment by ryangjchandler** [comment, marked as answer]
@jrmajor That's not supported with `Alpine.data` at the moment - for now, you're better off sticking to a global function or doing your setup inside of `x-init`.

**Reply by jrmajor** [reply]
Thank you!
